CMakeLists.txt

```diff
@@ -6,16 +6,18 @@ set(CRUNCH_EXE_NAME crunch)
 
 project(${CRUNCH_PROJECT_NAME})
 
-option(BUILD_SHARED_LIBS "Build crunchlib as shared library and link executables against it." OFF)
-option(BUILD_EXAMPLES "Build examples." OFF)
+option(BUILD_SHARED_LIBS "Link executables against shared libary" OFF)
```

Comment:
I think this should be given a name that more accurately describes what it actually does. Using a magic variable name doesn't seem like a good thing to me; it only creates confusion for the reader since you don't use the corresponding magical functionality.

Typo: `libary`

Reply:
If Crunch is integrated in a parent project that makes use of the [standardized `BUILD_SHARED_LIBS` variable name](https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html), this is the expected behaviour, even if the name sounds weird on Crunch local level.

The purpose of `BUILD_SHARED_LIBS` is, for a parent project, to build the parent project while linking to crunch shared library. Such parent project does not care about how crunch is built, the name of the variable is about how the parent project is built.

The `BUILD_SHARED_LIBS` variable is expected to be set in the parent project, not when building a Crunch standalone. This implementation of `BUILD_SHARED_LIBS` in Crunch is there to honor that variable if set by a parent project.

The important part of the cmake documentation is:

> so that each user of a project can decide if they want to build the project using shared or static libraries

Here, the “project” is a parent project linking against multiple libraries, crunch being one of them and inheriting the variable. And here, “build” is about the executable linking against `crnlib`.

The usage of `BUILD_SHARED_LIBS` comes from the fork from where I pulled that cmake implementation back in the time, I believe it's a good idea to not break that feature even if we don't use it ourselves. This is to ensure crunch still integrate with cmake projects not only using the single file header decompression library like we do, but also when linking against the whole `crnlib`.

Reply:
The change in any case breaks compatibility with anyone using the static crnlib (which I would imagine is most of them), because linking to `${CRUNCH_LIBRARY_NAME}` now gives them the shared one.

Other ways compatibility can potentially break:
- The toolchain can't build one or the other type of library, thus breaking the build.
- They are using some `FindLibrary` stuff and wanted the static library, but the shared library now exists and is preferred by default.

Reply:
> because linking to ${CRUNCH_LIBRARY_NAME} now gives them the shared one.

I updated the CMakeLists.txt in a way `${CRUNCH_LIBRARY_NAME}` is always the one to be linked against.

> The toolchain can't build one or the other type of library, thus breaking the build.

Can this really happen? They are built the same, only the final part (turning a `.o` into a `.a` or `.so`) differs.

> They are using some FindLibrary stuff and wanted the static library, but the shared library now exists and is preferred by default.

Hmm, isn't already true for any library provided by the distribution?

Reply:
> > The toolchain can't build one or the other type of library, thus breaking the build.
> 
> Can this really happen? They are built the same, only the final part (turning a `.o` into a `.a` or a `.so`) differs.

For example in NaCl, I don't there is such a thing as a shared library.


> > They are using some `FindLibrary` stuff and wanted the static library, but the shared library now exists and is preferred by default.
> 
> Hmm, isn't already true for any library provided by the distribution?

You mean like a Linux distribution? Yes.


These aren't major issues though; it seems OK to merge in this form.